core/src/syscall/host.rs

```diff
@@ -90,49 +88,23 @@ impl Syscall for SyscallHostCall {
 
     // read the input length from the next register
     let len = ctx.rt.register(Register::X12) as usize;
-    if len % 4 != 0 {
-      tracing::debug!("host system call input (a2/x12) not aligned to 4 bytes");
-      return Err(StatusCode::InvalidSyscallArgument);
-    }
 
     // `len` is denominated in number of bytes; we read words in chunks of four bytes
     // then convert into a byte array.
     let input = if len > 0 {
-      let input_words = ctx.slice(arg2, len / 4);
-      Some(
-        input_words
-          .into_iter()
-          .flat_map(|word| word.to_le_bytes())
-          .collect(),
-      )
-    } else {
-      None
-    };
-
-    // read the method name from the next register
-    let len = ctx.rt.register(Register::X14) as usize;
-    if len % 4 != 0 {
-      tracing::debug!("host system call input (a4/x14) not aligned to 4 bytes");
-      return Err(StatusCode::InvalidSyscallArgument);
-    }
-
-    let method_name_ptr = ctx.rt.register(Register::X13);
-
-    // `len` is denominated in number of bytes; we read words in chunks of four bytes
-    // then convert into a byte array.
-    let method = if len > 0 {
-      let words = ctx.slice(method_name_ptr, len / 4);
-      Some(
-        words
-          .into_iter()
-          .flat_map(|word| word.to_le_bytes())
-          .collect(),
-      )
+      // Round up the length to the nearest word boundary
+      let input_words = ctx.slice(arg2, (len + 3) / 4);
+      let input_bytes = input_words
+        .into_iter()
+        .flat_map(|word| word.to_le_bytes())
+        .collect::<Vec<u8>>();
+      // this removes any extra padding from the input
+      Some(input_bytes[..len].to_vec())
```

Comment:
How about
```suggestion
	.take(len)      // this removes any extra padding from the input
        .collect::<Vec<u8>>();
      Some(input_bytes)
```

to avoid creating 2 vectors.